lang/rust/avro/src/schema.rs

```diff
@@ -1441,6 +1441,9 @@ impl Parser {
             })?;
 
         for field in &fields {
+            if lookup.contains_key(&field.name) {
+                return Err(Error::FieldNameDuplicate(field.name.clone()));
+            }
             lookup.insert(field.name.clone(), field.position);
```

Comment:
I still think using the returned `Option` from `#insert()` is better than doing two lookups.
```suggestion
            if let Some(_old) = lookup.insert(field.name.clone(), field.position) {
                      return Err(Error::FieldNameDuplicate(field.name.clone()));
            }
```

Reply:
Ah, exactly.

Reply:
I'll manually fix.